### PR TITLE
One-time purchases

### PR DESCRIPTION
**Pull request by rebeccacremona**

### CODE REVIEW

File: perma-payments/perma_payments/tests/test_models.py [hunk 1]
```diff
@@ -666,6 +746,93 @@ def test_response_save_new_with_encrypted_full_response_ur(mocker, barebones_upd
...
+    assert response.full_response == b'someencryptedbytes'
+
+    # mocks called as expected
+    stringified.assert_called_once_with(spoof_django_post_object)
+    encrypted.assert_called_once_with(mocker.sentinel.stringified)
+
+
+# PurchaseRequestResponse
+
+def test_prr_inherits_from_outgiong_transaction():
```
Comment (bensteinberg) [inline perma-payments/perma_payments/tests/test_models.py]: Typo in function name here, doubt it matters

Comment (rebeccacremona) [reply perma-payments/perma_payments/tests/test_models.py]: thanks

File: perma-payments/perma_payments/tests/test_routes.py [hunk 2]
```diff
@@ -292,6 +337,202 @@ def test_index_other_methods(client, index):
...
+    response = client.post(acknowledge_purchase['route'])
+
+    # assertions
+    assert response.status_code == 200
+    prr.refresh_from_db()
+    assert prr.perma_acknowleged_at
+
+
+@pytest.mark.django_db
+def test_acknowledge_unacknowledgable_purchase(client, acknowledge_purchase, purchase_request_response_factory, mocker):
```
Comment (bensteinberg) [inline perma-payments/perma_payments/tests/test_routes.py]: I think I'd spell this `test_acknowledge_unacknowledgeable_purchase` but it certainly doesn't matter.

File: perma-payments/perma_payments/models.py [hunk 3]
```diff
@@ -642,3 +686,96 @@ def customer_pk(self):
...
+
+    related_request = models.OneToOneField(
+        PurchaseRequest,
+        related_name='purchase_request_response'
+    )
+    inform_perma = models.BooleanField(
+        default=False,
+        help_text='Should Perma be informed that this customer has successfully purchased more links?'
+    )
+    perma_acknowleged_at = models.DateTimeField(
```
Comment (bensteinberg) [inline perma-payments/perma_payments/models.py]: Typo, should be `perma_acknowledged_at` here and several other locations.

**Pull request merged**